### PR TITLE
#159987718 Implement pagination in the articles route

### PR DESCRIPTION
**Pull request by d-beloved**
#### What does this PR do?
- Have pagination working on the get all articles route

#### Description of Task to be completed?
- have articles broken into many pages of equal no. of articles
- if 200 articles are present in the app, it can be specified that 10 article should be returned per page and that will generate 20pages.

#### How should this be manually tested?
- clone the repo
- install nodejs and postgresdb in your local machine
- run ```npm install```
- insert a ```.env``` file according the ```.env-sample``` in the repo
- get the docs at ```localhost:3000/api-docs``` for info on the available routes
- insert articles into the app by first signing up on the app then adding articles following the documentation
- then hit ```localhost:3000/api/articles``` to view the number of articles returned.

#### What are the relevant pivotal tracker stories?
- 159987718

### CODE REVIEW

File: server/utils/articlesPaginate.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @returns {object} pagination - Pagination object
+ */
+
+const paginateArticle = ({
+  count = 0, rows = []
+}, currentPage, limit) => {
+  const totalRecords = count;
+  const totalPages = Math.ceil(totalRecords / limit);
+  const newRecipes = Object.assign({
```
Comment (lauragift21) [inline server/utils/articlesPaginate.js]: @d-beloved You might want to change the `newRecipes` name you used to make it align with the context.

**Review by lauragift21 — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "lint": "eslint ./ --fix",
     "migrate": "sequelize db:migrate:undo:all && sequelize db:migrate && sequelize db:seed:all",
     "pretest": "NODE_ENV=test npm run migrate",
-    "test": "NODE_ENV=test nyc mocha --compilers js:babel-register ./server/tests/ --timeout 5000ms --exit --no-deprecation",
+    "test": "NODE_ENV=test nyc mocha --compilers js:babel-register ./server/tests/* --timeout 5000ms --exit --no-deprecation",
```
Comment (AdmiralOnu) [inline package.json]: revert this to the line above. and add the missing test into one the corresponding tests/../index.js

File: server/controllers/article.js [hunk 3]
```diff
@@ -73,16 +74,24 @@ class ArticleController {
    * @description returns all article.
   */
   static getAll(req, res) {
```
Comment (AdmiralOnu) [inline server/controllers/article.js:76]: you need to write a test for this. I think this function returns the same two article each time

File: server/utils/articlesPaginate.js (hunk 1 above)
Comment (d-beloved) [reply server/utils/articlesPaginate.js]: Thanks, done

File: package.json (hunk 2 above)
Comment (d-beloved) [reply package.json]: Done, thanks

**Review by gonuoha — COMMENTED** [review]
ensure your .eslintrc file is updated. It should fix hound

File: server/controllers/article.js (hunk 3 above)
Comment (d-beloved) [reply server/controllers/article.js:76]: Alright, thanks

**Pull request closed without merging**